test/blackbox/common/DDSBlackboxTestsDiscovery.cpp

```diff
@@ -2405,3 +2407,94 @@ TEST(DDSDiscovery, proxy_data_assignment_operator_always_include_type_informatio
     // The second reader should match with the writer
     ASSERT_TRUE(readers.wait_discovery({0u, 1u}, std::chrono::seconds(3)));
 }
+
+/*!
+ * @test: regression test for redmine issue #23252
+ *
+ * This test checks that only one packet is sent to multicast (per interface)
+ * when a writer is matched with multiple readers.
+ * Typically, it is expected to send one via localhost and another via local lan interface.
+ */
+TEST(DDSDiscovery, multicast_only_one_packet_sent_when_multiple_multicast_readers_matched)
+{
+    using namespace eprosima::fastdds::dds;
+    using namespace eprosima::fastdds::rtps;
+    using namespace eprosima::fastdds::rtps;
+
+    std::vector<std::shared_ptr<PubSubReader<HelloWorldPubSubType>>> readers;
+    PubSubWriter<HelloWorldPubSubType> writer(TEST_TOPIC_NAME);
+
+    auto writer_test_transport = std::make_shared<test_UDPv4TransportDescriptor>();
+
+    const uint8_t n_readers = 5u;
+    uint8_t n_multicast_times_sent = 0u;
+
+    // Add new multicast locator with IP 239.255.0.4 and port 7900
+    DataReaderQos reader_qos;
+    eprosima::fastdds::rtps::Locator_t new_multicast_locator;
+    eprosima::fastdds::rtps::LocatorList multicast_locators;
+    eprosima::fastdds::rtps::IPLocator::setIPv4(new_multicast_locator, "239.255.0.4");
+    new_multicast_locator.port = 7900u;
```

Comment:
I would use a port outside the DDS port range, since black-box tests are run across all possible domain ids.

```suggestion
    new_multicast_locator.port = 6900u;
```